Update Terraform code for COOL multi-account environment

jsf9k opened the pull request:
## 🗣 Description

In this pull request I update this repo to work with our current implementation of the COOL multi-account environment.

## 💭 Motivation and Context

This code is to be deployed to the COOL.  Therefore it must work with our COOL multi-account environment.

## 🧪 Testing

I deployed these changes to production, and I verified that all the pre-commit hooks passed.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Add *.tfvars files to .gitignore

jsf9k committed:
Use the cool-terraform-backend profile for the TF backend

jsf9k committed:
Adjust backend and providers for COOL multi-account

jsf9k committed:
Add a policy to allow provisioning of the network resources

The policy is attached to the ProvisionAccount role.

jsf9k committed:
Rename repo, to be more consistent with what we have been doing lately

jsf9k committed:
Update README to agree with recent variable changes

jsf9k committed:
Remove the usage section since this is a deployment not a module

dav3r commented on provisionnetworking_policy.tf:
```diff
@@ -0,0 +1,23 @@
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "provisionnetworking_policy_doc" {
+  # This policy needs to be more restricted, obviously
```
🎗Reminder to create an issue to go back and restrict this policy (if this PR is merged and this policy is still wide open).

jsf9k replied:
See commit 5eb9f36.

dav3r reviewed (approved):
Looks great, just the one reminder... 🤓

jsf9k committed:
Merge remote-tracking branch 'skeleton-tf-module/develop' into improvement/update-for-cool-multiaccount

jsf9k committed:
Merge branch 'develop' into improvement/update-for-cool-multiaccount

mcdonnnj reviewed (approved):
I second @dav3r 's comment, but LGTM 👍

felddy reviewed (approved):
Looks good.  🎗

jsf9k committed:
Enumerate the permissions necessary to apply and destroy these resources

jsf9k committed:
Add depends_on clauses

This is for resources that require the ProvisionNetworking policy.

jsf9k committed:
Add a section to the README explaining initial targeted apply

Since Terraform does not yet support depends_on for modules, it is
necessary to run an initial partial apply (to attach the
ProvisionNetworking policy to the ProvisionAccount role) before
running a full terraform apply.

This is something that will be fixed in the future.  See
https://github.com/hashicorp/terraform/issues/17101 for details.

jsf9k committed:
Updates related to recent changes to distributed-subnets-tf-module

These changes are related to preferring for_each to count and
outputting actual resources instead of IDs.
